lib/assets/javascripts/cable.coffee.erb

```diff
@@ -5,8 +5,19 @@
   INTERNAL: <%= ActionCable::INTERNAL.to_json %>
 
   createConsumer: (url = @getConfig("url")) ->
-    new Cable.Consumer url
+    new Cable.Consumer websocketUrl(url)
 
   getConfig: (name) ->
     element = document.head.querySelector("meta[name='action-cable-#{name}']")
     element?.getAttribute("content")
+
+  websocketUrl = (url) ->
+    if url and not /^wss?:/i.test(url)
+      a = document.createElement("a")
+      a.href = url
+      # in internet explorer, the next line converts a relative url to an absolute one.
+      a.href = a.href
```

Comment:
I can't see that this actually converts or fixes anything. I tested in IE 10, 11 and Edge. And always got the same results with this test:

```js
window.location.toString()
=> "https://github.com/rails/actioncable"

a = document.createElement("a")

a.href = "/websocket"
a.href
=> "https://github.com/websocket"

a.href = a.href
a.href
=> "https://github.com/websocket"
```